airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/interpolated_string.py

```diff
@@ -37,6 +38,12 @@ def eval(self, config: Config, **kwargs: Any) -> Any:
         :param kwargs: Optional parameters used for interpolation
         :return: The interpolated string
         """
+        if self._just_string:
```

Comment:
Since this does a bit of caching magic, I'd love a comment explaining why you're doing this in eval specifically. I understand you need config to actually interpolate, so this is not insane, but a comment for posterity would be good.

Also, are there scenarios where the first interpolation would render the same string, but later interpolation would render something different? I can't think of it, but is there a world where a jinja expression renders itself / morphs conditionally on an input?

Reply:
+1. The change makes sense to me, but let's describe the rationale for posterity.

That being said, thank you for taking the time to contribute to Airbyte @szubster ! <3

Reply:
If there is any (which I doubt) then the output would still not look like input?
There are code-challenges, like https://en.wikipedia.org/wiki/Quine_(computing) but I doubt it's a real-life scenario.
And even if it is, then the logic is still correct :)